service/history/workflow/mutable_state_impl.go

```diff
@@ -4410,6 +4425,14 @@ func (ms *MutableStateImpl) ApplyWorkflowExecutionOptionsUpdatedEvent(event *his
 	if err := ms.updateVersioningOverride(attributes.GetVersioningOverride()); err != nil {
 		return err
 	}
+	if attributes.GetAttachedRequestId() != "" {
+		ms.attachRequestID(attributes.GetAttachedRequestId())
+	}
+	if len(attributes.GetAttachedCompletionCallbacks()) > 0 {
```

Comment:
We need to verify that the total number of attached callbacks on this workflow doesn't exceed the configured limit.

Reply:
I saw that you've added the check above, please make sure to have test coverage for this.